packages/titanium-docgen/generators/typescript_generator.js

```diff
@@ -182,7 +198,14 @@ class DocsParser {
 		const parentNamespaceName = namespaceParts.join('.');
 		let parentNamespace = null;
 		if (!this.tree.hasNamespace(parentNamespaceName)) {
-			let namespacePathFromRoot = [];
+			if (this.apis[parentNamespaceName]) {
```

Comment:
Yeah, there were some problems in `findOrCreateNamespace` method, but they were not actually caused by sorting.
I made changes in it and also removed pre-sorting.

> This check will always evaluate to true and the code below never runs.

But in the old implementation it also was never executed. This check always returned `true`:
```javascript
if (!this.tree.hasNamespace(parentNamespaceName)) {
```
because you manually created nodes for `Ti`, `Ti.Proxy`, `Ti.View` and `Ti.UI`.

Reply:
Basically this code was never-ever executed and now I completely removed it.

Reply:
Great, that fixes my main concert with the previous implementation.!